### DISCUSSION TITLE
Trame in a Public IP web server

### DISCUSSION THREAD

**Discussion by Rafaelrr5** [Kitware/trame#367] [Q&A] [answered]
Hey! I have just started with web servers on AWS Ubuntu and trame for a project in College and I would appreciate some help. Lately I've been trying to make my trame app run on a public web server with an Amazon EC2 Instance, and I've seen that trame-server only works on private ips. I already have the trame app with the trame server in a .py file and I wanna know if is there a way that I can just make it public with only trame server or if I have to configure another kind of server that will publish my trame app. I've tried some little things with NGINX but nothing succeeded (I don't have much experience with it either).
I've seen some things about wslink, which maybe is the way to go, and caprover, that maybe is not (in my opinion) the way I really wanna go. Flask is an option too but I didn't started looking for ways with it cause I think that's maybe there are another good options for me to look into and I hope someone can help me with that. Thanks!

**Comment by jourdain** [comment]
If you build a docker image for your trame app, you can then publish it to caprover. There is no limitation about public/private IP. 

You can look at that example https://github.com/Kitware/trame/tree/master/examples/deploy/docker/SingleFile

**Reply by Rafaelrr5** [reply]
But then I'd have the caprover costs, right? Is there a way I can have it on my on server? I already have the configured AWS EC2 instance available.

**Reply by jourdain** [reply]
What do you mean by caprover cost? Caprover is free. Just install/run it (run the docker command) on your EC2 instance.

**Reply by jourdain** [reply]
https://caprover.com/

**Reply by Rafaelrr5** [reply]
I read "In standard installation, CapRover has to be installed on a machine with a public IP address. If you need help with Public IP, see [Server & Public IP address](https://caprover.com/docs/server-purchase.html). This will cost you as low as $5 a month.", but I guess the docker install is really free so I may try it

**Reply by jourdain** [reply]
What they are talking about is the cost of the hosting service, like your EC2 that you already have.

**Comment by jourdain** [comment]
Where did you see that? Because that is not true.

> I've seen that trame-server only works on private ips.

Also if you just want one app, caprover might be overkill but still pretty simple to use and setup. So for the one app, just create a docker image of your app, and run that docker image.

**Reply by Rafaelrr5** [reply]
I understood it that way on some part of trame guide. Is there a specific server configuration on trame server so I can get out of localhost and private IP?
If not, I may try the the caprover then

**Reply by jourdain** [reply, marked as answer]
Just add the arg `--host 0.0.0.0`, but that won't properly work for several users. 

Hence the docker setup that does not require caprover, but caprover will manage the SSL and multi-app part on top of the multi-user managed by the docker image itself.

**Reply by Rafaelrr5** [reply]
Perfect, that should do it then! I'll try it. Tysm
